cli/cmd/inject.go

```diff
@@ -123,13 +123,15 @@ func (rt resourceTransformerInject) transform(bytes []byte) ([]byte, []inject.Re
 	if len(rt.proxyOutboundCapacity) > 0 {
 		conf = conf.WithProxyOutboundCapacity(rt.proxyOutboundCapacity)
 	}
-	nonEmpty, err := conf.ParseMeta(bytes)
+
+	report, err := conf.ParseMetaAndYAML(bytes)
```

Comment:
Now that this call returns a singular report, you can remove the `len(reports)` check on line 160 in this file, similar to the change you're making in webhook.go:

```diff
diff --git a/cli/cmd/inject.go b/cli/cmd/inject.go
index 18de30a5..c4cf0dcc 100644
--- a/cli/cmd/inject.go
+++ b/cli/cmd/inject.go
@@ -156,11 +156,7 @@ func (rt resourceTransformerInject) transform(bytes []byte) ([]byte, []inject.Re
 	if patchJSON == nil {
 		return bytes, reports, nil
 	}
-	// TODO: refactor GetPatch() for it to return just one report item
-	if len(reports) > 0 {
-		r := reports[0]
-		log.Infof("patch generated for: %s", r.ResName())
-	}
+	log.Infof("patch generated for: %s", report.ResName())
 	log.Debugf("patch: %s", patchJSON)
 	patch, err := jsonpatch.DecodePatch(patchJSON)
 	if err != nil {
```